modules/templates/nimzo/config.py

```diff
@@ -0,0 +1,1187 @@
+# -*- coding: utf-8 -*-
+
+try:
+    # Python 2.7
+    from collections import OrderedDict
+except:
+    # Python 2.6
+    from gluon.contrib.simplejson.ordered_dict import OrderedDict
+
+from gluon import current
+from gluon.html import *
+from gluon.storage import Storage
+
+
+# Is not entirely done
+def semantic_formstyle(id, label, controls, help):
```

Comment:
This is getting called with other args, so maybe add some catch-all args, like *args, **kwargs.  You don't need to do anything with those, just have them so there is a place for the extra args to go.  If you want to watch what gets sent in that you don't currently handle, you can print what ends up in args and kwargs.  (I'd suggest printing to stderr rather than stdout -- Web2py does not always show things that go to stdout.  There are a bunch of "def _debug" functions in modules/s3 as examples.  I thought we had a shared "print to stderr" function, but don't see it any more.)

When I went to this URL in my local instance, I got a 500 error:
http://127.0.0.1:8000/gci/inv/warehouse

Here is the trace -- it says an arg "hidden" was passed to semantic_formstyle:
Traceback (most recent call last):
  File "C:\web2py\gluon\restricted.py", line 224, in restricted
    exec ccode in environment
  File "C:/web2py/applications/gci/controllers/inv.py", line 1695, in <module>
  File "C:\web2py\gluon\globals.py", line 393, in <lambda>
    self._caller = lambda f: f()
  File "C:/web2py/applications/gci/controllers/inv.py", line 388, in warehouse
    rheader = s3db.inv_rheader,
  File "C:/web2py/applications/gci/models/00_utils.py", line 224, in s3_rest_controller
    output = r(**attr)
  File "applications\gci\modules\s3\s3rest.py", line 637, in __call__
    output = self.resource.crud(self, **attr)
  File "applications\gci\modules\s3\s3rest.py", line 1704, in __call__
    output = self.apply_method(r, **attr)
  File "applications\gci\modules\s3\s3crud.py", line 115, in apply_method
    output = self.select(r, **attr)
  File "applications\gci\modules\s3\s3crud.py", line 1204, in select
    form = self.create(r, **attr).get("form", None)
  File "applications\gci\modules\s3\s3crud.py", line 376, in create
    format=representation)
  File "applications\gci\modules\s3\s3forms.py", line 421, in __call__
    buttons = buttons)
  File "C:\web2py\gluon\sqlhtml.py", line 1220, in __init__
    inp = field.widget(field, default)
  File "applications\gci\modules\s3\s3widgets.py", line 4602, in __call__
    visible_components = self._layout(components, map_icon=map_icon)
  File "applications\gci\modules\s3\s3widgets.py", line 4922, in _layout
    hidden=hidden,
TypeError: semantic_formstyle() got an unexpected keyword argument 'hidden'